llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -658,6 +747,19 @@ void GCNScheduleDAGMILive::runSchedStages() {
         continue;
       }
 
+      if (GCNTrackers) {
+        GCNDownwardRPTracker *DownwardTracker = S.getDownwardTracker();
+        GCNUpwardRPTracker *UpwardTracker = S.getUpwardTracker();
+        GCNRPTracker::LiveRegSet *RegionLiveIns =
+            &LiveIns[Stage->getRegionIdx()];
+
+        reinterpret_cast<GCNRPTracker *>(DownwardTracker)
```

Comment:
This looks criminal. You're forcing a call of a base class constructor for an object of derived type. Why this is needed?

Reply:
Sorry I meant "reset" method, not constructor.  Basically you're leaving the derived object in undefined state.

Reply:
It was needed because the overloaded resets in GCNUpwardRPTracker/GCNDownwardRPTracker hide the GCNRPTracker::reset , fixed in https://github.com/llvm/llvm-project/pull/93088